src/OrchardCore.Modules/OrchardCore.Contents/Workflows/Activities/ContentActivity.cs

```diff
@@ -27,6 +28,18 @@ protected ContentActivity(IContentManager contentManager, IWorkflowScriptEvaluat
 
         protected IWorkflowScriptEvaluator ScriptEvaluator { get; }
 
+        protected bool FromContentDriver { get; private set; }
+
+        protected bool FromContentHandler { get; private set; }
+
+        protected string OriginalCorrelationId { get; private set; }
+
+        protected bool AsCorrelatedContentDriver { get; private set; }
+
+        protected bool AsCorrelatedContentHandler { get; private set; }
+
+        protected bool AsCorrelatedContentDriverOrHandler { get; private set; }
```

Comment:
Ok @jtkech I had some time to have a look.

It feels complicated (but it is a complicated scenario), and I can see it all works :)

Why do these properties need to be protected? 
From what I can see they are all set internally in the `ContentActivity` by various evaluations where you calculate how the inputs were recieved?

I don't mind that they are protected, it just confused me when I was trying to understand what they did, because I thought maybe one of the implementations had to set them, but then it didn't seem that is the case.

**Edit** Is it just so the workflows can return the `Noop` outcome? That makes sense then. (maybe a comment?)

Reply:
@deanmarcussen yes it is complicated because a Content Task may be executed inline e.g. as a content handler. This when inline after a Content Event triggered by the `OC.Contents.Workflows.Handlers ContentsHandler` that calls `TriggerWorkflowEventAsync` which sets the WF context input and correlationId by using the related content item. Inline meaning that between the Content Event and Content Task there is no waiting activity, meaning that the Content Task is executed in the same scope and acts as a content handler and should not call content manager methods.

Note: I found a solution by triggering the WF in a deferred task, so in another scope, and it was much more simple, but then e.g. a Notify Task after a Content Event was not working because the controller redirects before the WF execution. So i didn't use this and let execute inline a WF that is triggered by a Content Event.

> Why do these properties need to be protected?

Yes, these properties are mainly managed here, particularly though `OnInputReceivedAsync()` to see if the WF has been triggered from e.g. a content handler and to hold the `OriginalCorrelationId`. Then i made them protected so that they can be used by derived classes, see in `UpdateContentTask.cs` where `AsCorrelatedContentHandler` and `AsCorrelatedContentDriverOrHandler` are used and more frequently.

Reply:
Cool, yes, this is what I thought you were doing.

If there is an opportunity for simplification that would be great :)

`Inline` or `Inscope` make a bit more sense to me when reading this, rather than `AsCorrelateContentDriverOrHandler`

Is there a way the naming of these methods could be more generic, i.e. `Inline` or `Inscope`

This `AsCorrelatedContentDriverOrHandler` feels so tied to a `Driver` or `Handler` (and one day I guess there will be GraphQL mutations to take into account here too, for example).

Not trying to make more work for you here, and maybe my suggestions make no sense, so only if you agree.

Reply:
@deanmarcussen thanks for the feedback

> If there is an opportunity for simplification that would be great :)

I'll try, maybe i can remove the distinction between driver / handler, maybe some simplifications after the recent changes on yesSql side. I will rework on it.

> Is there a way the naming of these methods could be more generic, i.e. Inline or Inscope

Thanks for the suggestions, i like these names, i will think about it

> This AsCorrelatedContentDriverOrHandler feels so tied to a Driver or Handler (and one day I guess there will be GraphQL mutations to take into account here too, for example).

It is already related to admin controller, api controller, and yes one day maybe to graphQL, but normally always through content manager methods **and then content handlers**. About drivers it's only related to the WF `UserTaskEvent` that let you add user buttons when editing the related item, and that trigger an `UserTaskEvent` through a specific driver.